packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp

```diff
@@ -0,0 +1,452 @@
+/**
+ * @file SubmitterClientExt.h
+ * @brief Header file for the SubmitterClientExt class.
+ */
+#include "submitter/SubmitterClient.h"
+
+#include <future>
+#include <sstream>
+#include <string>
+
+#include "submitter_common.pb.h"
+#include "submitter_service.grpc.pb.h"
+#include "sessions_common.pb.h"
+#include "sessions_service.grpc.pb.h"
+#include "objects.grpc.pb.h"
+
+
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::Status;
+using armonik::api::grpc::v1::submitter::CreateSessionRequest;
+using armonik::api::grpc::v1::submitter::CreateSessionReply;
+using armonik::api::grpc::v1::submitter::Submitter;
+using armonik::api::grpc::v1::TaskOptions;
+using armonik::api::grpc::v1::TaskRequest;
+using armonik::api::grpc::v1::ResultRequest;
+using namespace armonik::api::grpc::v1::submitter;
+
+/**
+ * @brief Construct a new Submitter Client:: Submitter Client object
+ * 
+ * @param stub the gRPC client stub 
+ */
+SubmitterClient::SubmitterClient(std::unique_ptr<armonik::api::grpc::v1::submitter::Submitter::StubInterface> stub) {
+  stub_ = std::move(stub);
+}
+
+/**
+ * @brief Create a new session.
+ * @param partition_ids The partitions ids.
+ * @param default_task_options The default task options.
+ */
+std::string SubmitterClient::create_session(TaskOptions default_task_options,
+                                            const std::vector<std::string>& partition_ids = {"default"})
+{
+  CreateSessionRequest request;
+  *request.mutable_default_task_option() = default_task_options;
+  for (const auto& partition_id : partition_ids)
+  {
+    request.add_partition_ids(partition_id);
+  }
+  CreateSessionReply reply;
+  
+  Status status = stub_->CreateSession(&context_, request, &reply);
+  if (!status.ok())
+  {
+    std::stringstream message;
+    message << "Error: " << status.error_code() << ": " << status.error_message()
+            << ". details : " << status.error_details() << std::endl;
+    std::cout << "CreateSession rpc failed: " << std::endl;
+    throw std::runtime_error(message.str().c_str());
+  }
+  return reply.session_id();
+}
+
+/**
+ * @brief Cancel a session.
+ *
+ * @param session_id The session id.
+ */
+bool SubmitterClient::cancel_session(std::string_view session_id)
+{
+  std::shared_ptr<Channel> channel = grpc::CreateChannel(
+      "172.30.39.223:5001",
+                                grpc::InsecureChannelCredentials());
+  auto stub = armonik::api::grpc::v1::sessions::Sessions::NewStub(channel);
```

Comment:
You should not recreate a new channel here. Plus, the channel shall not be hardcoded.

Maybe you could remove the cancel_session rpc for now, and add it back in a new PR where you will tackle the multiple stub problem.